### DISCUSSION TITLE
Export in CSV is broken with multiline notes?

### DISCUSSION THREAD

**Discussion by e4rthdog** [dani-garcia/vaultwarden#4462] [General]
Export in CSV seems to have problem in general. Many of my vault items have multiline notes. when i export to CSV the file is messed up.

I want to have a password protected PDF with my vault items and in order to do that i need to get the vault items somehow.

Any ideas?

**Comment by stefan0xC** [comment]
> when i export to CSV the file is messed up.

As far as I've tested it the lines inside the the multiline notes have only line feeds (`\n`) (and are also escaped in double quotes), while the csv rows end with a proper [CRLF](https://developer.mozilla.org/en-US/docs/Glossary/CRLF) (`\r\n`) (cf. [RFC 4180](https://datatracker.ietf.org/doc/html/rfc4180)). So it might just be an issue with the way you view the files?

If you run `cat --show-ends bitwarden_export.csv` it should look like this:

```txt
folder,favorite,type,name,notes,fields,reprompt,login_uri,login_username,login_password,login_totp^M$
,,login,test,,,0,,user1,password1,^M$
,,login,test,"line 1$
line 2$
line 3",,0,,user2,password2,^M$
,,note,test note,"line 1$
line 2$
line 3",,0,,,,⏎
```

**Reply by e4rthdog** [reply]
Yeah i think it looks like you say:
![image](https://github.com/dani-garcia/vaultwarden/assets/1651764/78ed134e-72d3-4c4d-90b5-ab203420db6b)

I am opening this via Excel. It should work , i will look into getting Excel to properly import it.

**Reply by stefan0xC** [reply]
I've only tested with LibreOffice where you can set the string delimiter when importing CSV. Not sure about Excel.

**Reply by e4rthdog** [reply]
Just checked it. Excel is workign ok if you go from the `Data` menu and not the import wizard.

![image](https://github.com/dani-garcia/vaultwarden/assets/1651764/e2db870e-82ac-4181-bfad-f00d1404550f)
